test/test_api.py

```diff
@@ -289,7 +299,8 @@ def test_it_does_not_return_restricted_affiliations(test_client: FlaskClient, ye
     profile.add_affiliation(affiliation)
     profile.add_affiliation(affiliation2)
 
-    db.session.commit()
+    with patch('profiles.orcid.request'):
```

Comment:
smells of a duplicate concept, can these two lines be extracted to cater for whatever else we want to mock?

Reply:
It's preventing HTTP calls from the `OrcidClient` that is used through the registered `webhook_maintainer` in the `create_app` call. This was the cleanest way to prevent the calls (just mocking it when `commit` is called). I tried mocking out the passed in `OrcidClient` in the `create_app` call (this needs to be done in the `conftest` as the `app` has a session scope) but it got messy and felt like it was muddying the whole `app` creation which had knock-on effects on other tests. Also sometimes you might not want to mock this and intercept or inspect the HTTP call.

Open to suggestions.

Reply:
Just meant to extract:
```
def _commit():
    with patch('profiles.orcid.request'):
         db.session.commit()
```
in this file

Reply:
Ah ok, well I didn't want to start pulling snippets into functions (in local files) as we have not done this throughout the test suite, everything has been structured as fixtures in `conftest` and injected.

That said I agree with the suggestion, I have made the change. If we start requiring this in other test files it may need to be moved/refactored.